site/zenodo_rdm/subcommunities/request.py

```diff
@@ -113,8 +124,180 @@ class ZenodoSubCommunityRequest(SubCommunityRequest):
 
     available_actions = {
         "delete": actions.DeleteAction,
-        "create": SubcommunityCreateAction,
         "cancel": actions.CancelAction,
+        # Custom implemented actions
+        "create": SubcommunityCreateAction,
         "accept": SubcommunityAcceptAction,
         "decline": DeclineSubcommunity,
     }
+
+
+class SubcommunityInvitationCreateAction(actions.CreateAction):
+    """Represents an accept action used to accept a subcommunity."""
+
+    def execute(self, identity, uow):
+        """Execute approve action."""
+        parent = self.request.created_by.resolve()
+        if not parent.children.allow:
+            raise ValidationError("Assigned parent is not allowed to be a parent.")
+        super().execute(identity, uow)
+
+
+class SubCommunityInvitationAcceptAction(AcceptSubcommunityInvitation):
+    """Represents an accept action used to accept a subcommunity.
+
+    Zenodo re-implementation of the accept action, to also move the records.
+    """
+
+    def _get_community_records(self, community_id):
+        """Get the records of a community."""
+        return current_community_records_service.search(
+            system_identity, community_id=community_id
+        )
```

Comment:
major/shelve: this method only returns the first 25 results from the record search, so the code below doesn't "bulk add" all the records of the community.

This actually is a bug even in our subcommunity inclusion request logic above, which means that when we accept subcommunity requests, this doesn't add ALL the records of the community to the parent community. 

It's relatively low impact though, since most communities don't have that many records (or probably have no records at all, since they've just being created). For the new request type though, this will not be the case, so this has to be fixed relatively quick.